pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,327 @@
+		return nil
+	}
+	ds, err := g.collectDataSource.GetDataSources(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve datasource: %w", err)
+	}
+
+	for _, grds := range ds.GithubReleaseDataSources {
+		r, t, err := ParseGithubReleaseDataSource(grds)
+		if err != nil {
+			logger.Warnf("unable to parse github datasource: %w", err)
+			continue
+		}
+		g.repoToReleaseTags[*r] = append(g.repoToReleaseTags[*r], t)
+	}
+
+	for _, gds := range ds.GitDataSources {
+		r, t, err := ParseGitDataSource(gds)
+		if err != nil {
+			logger.Warnf("unable to parse git datasource: %w", err)
+		}
+		g.repoToReleaseTags[*r] = append(g.repoToReleaseTags[*r], t)
+	}
+
+	return nil
+}
+
+func (g *githubCollector) fetchAssets(ctx context.Context, owner string, repo string, tags []TagOrLatest, docChannel chan<- *processor.Document) {
+	logger := logging.FromContext(ctx)
+	var releases []client.Release
+	for _, gitTag := range tags {
+		var release *client.Release
+		var err error
+		switch t := gitTag.(type) {
+		case *Latest:
+			release, err = g.client.GetLatestRelease(ctx, owner, repo)
+		case *Tag:
+			release, err = g.client.GetReleaseByTag(ctx, owner, repo, t.Value)
+		}
+		if err != nil {
+			logger.Warnf("unable to fetch release: %w", err)
+			continue
+		}
+		releases = append(releases, *release)
+	}
+
+	for _, release := range releases {
+		g.collectAssetsForRelease(ctx, release, docChannel)
+	}
+}
+
+func (g *githubCollector) collectAssetsForRelease(ctx context.Context, release client.Release, docChannel chan<- *processor.Document) {
+	logger := logging.FromContext(ctx)
+	for _, asset := range release.Assets {
+		if checkSuffixes(asset.URL, g.assetSuffixes) {
+			content, err := g.client.GetReleaseAsset(asset)
+			if err != nil {
+				logger.Warnf("unable to download asset: %w", err)
+				continue
+			}
+			doc := &processor.Document{
+				Blob:   content.Bytes,
+				Type:   processor.DocumentUnknown,
+				Format: processor.FormatUnknown,
+				SourceInformation: processor.SourceInformation{
+					Collector: GithubCollector,
+					Source:    content.Name,
+				},
+			}
+			docChannel <- doc
+		}
+	}
+}
+
+func checkSuffixes(name string, suffixes []string) bool {
+	for _, suffix := range suffixes {
+		if strings.HasSuffix(name, suffix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// TODO (mlieberman85): Parse* below are helpers both used by the collector as well as others
+// to parse data sources into internal structs used by various components.
+// This should be moved into another location for helpers.
+
+func ParseGithubReleaseDataSource(source datasource.Source) (*client.Repo, TagOrLatest, error) {
+	u, err := url.Parse(source.Value)
+	if err != nil {
+		return nil, nil, err
+	}
+	if u.Scheme != "https" {
+		return nil, nil, fmt.Errorf("invalid github url scheme: %v", u.Scheme)
+	}
+	if u.Host != "github.com" {
+		return nil, nil, fmt.Errorf("invalid github host: %v", u.Host)
+	}
+	// [1:] to ignore leading slash
```

Comment:
it would be helpful to have a comment here to define what the different array indices refer to or the cases you think are valid returns here, it'll make it easier to compare to documentation if something decides to change upstream

Reply:
Sure.